src/Model/Item.php

```diff
@@ -2002,13 +2002,23 @@ private static function getLanguage(array $item): string
 			return $item['language'];
 		}
 
+		$transmitted = [];
+		foreach ($item['transmitted-languages'] ??  [] as $language) {
+			$transmitted[$language] = 0;
+		}
+
 		if (!in_array($item['gravity'], [self::GRAVITY_PARENT, self::GRAVITY_COMMENT]) || empty($item['body'])) {
-			return '';
+			return empty($transmitted) ? '' : json_encode($transmitted);
 		}
 
 		$languages = self::getLanguageArray($item['title'] . ' ' . ($item['content-warning'] ?? '') . ' ' . $item['body'], 3, $item['uri-id'], $item['author-id']);
 		if (empty($languages)) {
-			return '';
+			return empty($transmitted) ? '' : json_encode($transmitted);
```

Comment:
Why is the return of this function either a regular string or a JSON-encoded string? Why not always the latter?

Reply:
When no language data is present, `empty($item['language])` still should be true. This is not the case, when we would use `json_encode` on an empty array.

Reply:
Next question: why not return a PHP array and let whatever process downstream JSON encode for storage?

Reply:
That function is only used to store the data in the `language` field in the `post-content` table. We can't change that format, since otherwise we had to change all history data as well.

Reply:
I get that, but instead of doing `$field = self::getLanguages()` I believe we should do `$field = json_encode(self::getLanguages())`, meaning that the JSON encoding should be done only once just before saving, not at multiple points in the method itself which gives an inconsistent return value.

Reply:
Then we still had to check for an empty value and had to set the field to `""`.

Reply:
No, the field can be `[]`, the JSON encoding for an empty array, and when the field is read and decoded, the value will be falsy.

Reply:
We are using this `if (!empty($item['language']))` on several places. Also the historic data is created that way. So I don't want to change anything here.

Reply:
There are two ways the item language is used:
- In the database, where it's stored as a JSON array
- In the code, where it's used as a regular PHP array

If `$item['language']` is a JSON-encoded string at any point that isn't just before creating/updating an item or just after retrieving a item from the database, then we're doing this wrong and it leads to inconsistent return values for methods like `Item::getLanguages()`.